### PR TITLE
Mango: Add first working version

### PR DESCRIPTION
**Pull request by goldbattle**
Has browse, chapters, and searching.
The API endpoint is just the whole entire library, so searching is done locally.
Requires login authentication, so will need to view the webview first to login, then everything works as expected.
https://github.com/hkalexling/Mango/

### CODE REVIEW

File: src/all/mango/build.gradle [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    extName = 'Mango'
+    pkgNameSuffix = 'all.mango'
+    extClass = '.Mango'
+    extVersionCode = 1
+    libVersion = '1.2'
+}
+
+dependencies {
+    implementation 'io.reactivex:rxandroid:1.2.1'
+    implementation 'io.reactivex:rxjava:1.3.6'
```
Comment (CarlosEsco) [inline src/all/mango/build.gradle:14]: are these reactive libraries actually needed?

File: src/all/mango/src/eu/kanade/tachiyomi/extension/all/mango/Mango.kt [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+        val queryLower = query.toLowerCase()
+        val mangas = popularMangaParse(response).mangas
+        val exactMatch = mangas.firstOrNull { it.title.toLowerCase() == queryLower }
+        if (exactMatch != null) {
+            return MangasPage(listOf(exactMatch), false)
+        }
+
+        // Text distance algorithms
+        // Code is originally from here:
+        // https://github.com/CarlosEsco/Neko/blob/987d1485203a1ec1b5037dcf9b4bdea24f2cf8a3/app/src/main/java/eu/kanade/tachiyomi/source/online/MergeSource.kt
```
Comment (CarlosEsco) [inline src/all/mango/src/eu/kanade/tachiyomi/extension/all/mango/Mango.kt:95]: don't need this comment

File: src/all/mango/src/eu/kanade/tachiyomi/extension/all/mango/Mango.kt [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+            apiCookies = ""
+            throw Exception("Login Likely Failed. Try Refreshing.")
+        }
+        return result["entries"].asJsonArray.map { obj ->
+            SChapter.create().apply {
+                name = obj["display_name"].asString
+                url = "/page/${obj["title_id"].asString}/${obj["id"].asString}/${obj["pages"].asString}/"
+                date_upload = 1000L * obj["mtime"].asLong
+                // chapter_number = name.replace(Regex("[^0-9]"), "").toFloat()
+                // chapter_number = obj["mtime"].asLong.toFloat()
```
Comment (CarlosEsco) [inline src/all/mango/src/eu/kanade/tachiyomi/extension/all/mango/Mango.kt:154]: remove comments

**Review by CarlosEsco — CHANGES_REQUESTED** [review]

File: src/all/mango/build.gradle (hunk 1 above)
Comment (goldbattle) [reply src/all/mango/build.gradle]: Nope :)

**Pull request merged**